kahuna/public/js/downloader/downloader.js

```diff
@@ -5,6 +5,21 @@ import template from './downloader.html!text';
 
 export const downloader = angular.module('gr.downloader', []);
 
+// blob URLs have a max size of 500MB - https://github.com/eligrey/FileSaver.js/#supported-browsers
+const max_blob_size = 500 * 1024 * 1024;
+
+const bytesToSize = (bytes) => {
+    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+    if (bytes === 0) {
+        return 'n/a';
+    }
+    const i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)));
```

Comment:
Not needed, `Math.floor` already returns an int...

I've also refactored this function a little to make it simpler, returning fast when `bytes === 0`.